src/renderer/routes/route-path-parameters.injectable.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import { matchPath } from "react-router";
+import currentPathInjectable from "./current-path.injectable";
+import type { Route } from "../../common/front-end-routing/route-injection-token";
+
+const routePathParametersInjectable = getInjectable({
+  id: "route-path-parameters",
+
+  instantiate: (di, route: Route<unknown>) => {
+    const currentPath = di.inject(currentPathInjectable);
+
+    // TODO: Reuse typing from route for accuracy
+    return computed((): Record<string, string> => {
+      const match = matchPath(currentPath.get(), {
```

Comment:
I don't like this design, too defensive in nature. Instead I think it would be better to have these types cascade once matched.

Reply:
I don't quite understand what you mean here, but after this PR, ideas like this will sure start living in increasingly more fertile grounds, as they will be protected by the behavioural unit tests, and thus be free to be refactored and improved on over and over again.